src/pkgcheck/checks/eclass.py

```diff
@@ -98,6 +133,38 @@ def check_pre_inherits(self, pkg, inherits):
                     yield MisplacedEclassVar(
                         var_name, line=line, lineno=lineno+1, pkg=pkg)
 
+    def check_deprecated(self, pkg, inherits):
+        """Check for usage of @DEPRECATED variables or functions."""
+        deprecated_variables = {}
+        deprecated_functions = {}
+        # determine if any inherited eclasses have @DEPRECATED variables or functions
+        for eclasses, _ in inherits:
+            for eclass in eclasses:
+                for var in self.eclass_cache[eclass].variables:
+                    if var.deprecated:
+                        deprecated_variables[var.name] = var.deprecated
+                for func in self.eclass_cache[eclass].functions:
+                    if func.deprecated:
+                        deprecated_functions[func.name] = func.deprecated
+
+        # scan for usage of @DEPRECATED variables
+        if deprecated_variables:
+            for node, _ in bash.var_assign_query.captures(pkg.tree.root_node):
```

Comment:
I'd probably use `var_query` here instead of `var_assign_query` as I imagine some deprecated, eclass variables can be used without assigning to them (assuming this doesn't add false positives for whatever reason).